diff --git a/qmic-qt/src/DialogSettings.cpp b/qmic-qt/src/DialogSettings.cpp
index bf6b5cb..3dc9cf8 100644
--- a/qmic-qt/src/DialogSettings.cpp
+++ b/qmic-qt/src/DialogSettings.cpp
@@ -92,6 +92,10 @@ DialogSettings::DialogSettings(QWidget * parent) : QDialog(parent), ui(new Ui::D
   ui->cbNativeFileDialogs->setChecked(Settings::nativeFileDialogs());
   ui->cbNativeFileDialogs->setToolTip(tr("Check to use Native/OS file dialog, uncheck to use Qt's"));
   ui->cbShowLogos->setChecked(Settings::visibleLogos());
+#ifdef _GMIC_QT_DISABLE_LOGO_
+  ui->line->setVisible(false);
+  ui->cbShowLogos->setVisible(false);
+#endif
   ui->sbPreviewTimeout->setValue(Settings::previewTimeout());
   ui->cbPreviewZoom->setChecked(Settings::previewZoomAlwaysEnabled());
   ui->cbNotifyFailedUpdate->setChecked(Settings::notifyFailedStartupUpdate());
diff --git a/qmic-qt/digiKam/update_gmicqt.txt b/qmic-qt/digiKam/update_gmicqt.txt
index 709e0da..77430fb 100644
diff --git a/qmic-qt/src/Settings.cpp b/qmic-qt/src/Settings.cpp
index 0640f04..3611def 100644
--- a/qmic-qt/src/Settings.cpp
+++ b/qmic-qt/src/Settings.cpp
@@ -121,7 +121,11 @@ void Settings::load(UserInterfaceMode userInterfaceMode)
 
 bool Settings::visibleLogos()
 {
+#ifdef _GMIC_QT_DISABLE_LOGO_
+  return false;
+#else
   return _visibleLogos;
+#endif
 }
 
 void Settings::setVisibleLogos(bool on)
