diff --git a/src/DialogSettings.cpp b/src/DialogSettings.cpp
index a30c403..1359a9d 100644
--- a/src/DialogSettings.cpp
+++ b/src/DialogSettings.cpp
@@ -109,6 +109,7 @@ DialogSettings::DialogSettings(QWidget * parent) : QDialog(parent), ui(new Ui::D
   connect(ui->outputMessages, QOverload<int>::of(&QComboBox::currentIndexChanged), this, &DialogSettings::onOutputMessageModeChanged);
   connect(ui->cbNotifyFailedUpdate, &QCheckBox::toggled, this, &DialogSettings::onNotifyStartupUpdateFailedToggle);
 
+#ifndef _GMIC_USE_HOSTED_SETTINGS_
 #if QT_VERSION_GTE(6, 0, 0)
   ui->cbHighDPI->hide();
   ui->labelHighDPI->hide();
@@ -116,6 +117,10 @@ DialogSettings::DialogSettings(QWidget * parent) : QDialog(parent), ui(new Ui::D
   ui->cbHighDPI->setChecked(Settings::highDPIEnabled());
   connect(ui->cbHighDPI, &QCheckBox::toggled, this, &DialogSettings::onHighDPIToggled);
 #endif
+#else
+  ui->cbHighDPI->hide();
+  ui->labelHighDPI->hide();
+#endif
 
   ui->languageSelector->selectLanguage(Settings::languageCode());
   ui->languageSelector->enableFilterTranslation(Settings::filterTranslationEnabled());
diff --git a/src/GmicQt.cpp b/src/GmicQt.cpp
index 3af90b3..77b1559 100644
--- a/src/GmicQt.cpp
+++ b/src/GmicQt.cpp
@@ -545,6 +545,7 @@ namespace
 
 void configureApplication()
 {
+#ifndef _GMIC_USE_HOSTED_SETTINGS_
   QCoreApplication::setOrganizationName(GMIC_QT_ORGANISATION_NAME);
   QCoreApplication::setOrganizationDomain(GMIC_QT_ORGANISATION_DOMAIN);
   QCoreApplication::setApplicationName(GMIC_QT_APPLICATION_NAME);
@@ -554,6 +555,7 @@ void configureApplication()
     QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
   }
 #endif
+#endif
 }
 
 void disableModes(const std::list<GmicQt::InputMode> & disabledInputModes, //
diff --git a/src/HeadlessProcessor.cpp b/src/HeadlessProcessor.cpp
index de6dc90..a0f140e 100644
--- a/src/HeadlessProcessor.cpp
+++ b/src/HeadlessProcessor.cpp
@@ -267,7 +267,9 @@ void HeadlessProcessor::endApplication(const QString & errorMessage)
   if (!errorMessage.isEmpty()) {
     Logger::error(errorMessage);
   }
+#ifndef _GMIC_USE_HOSTED_SETTINGS_
   QCoreApplication::exit(!errorMessage.isEmpty());
+#endif
 }
 
 } // namespace GmicQt
