diff --git a/src/FilterSelector/FiltersPresenter.cpp b/src/FilterSelector/FiltersPresenter.cpp
index 9e43caa..e021194 100644
--- a/src/FilterSelector/FiltersPresenter.cpp
+++ b/src/FilterSelector/FiltersPresenter.cpp
@@ -90,6 +90,7 @@ void FiltersPresenter::rebuildFilterViewWithSelection(const QList<QString> & key
   _filtersView->clear();
   _filtersView->disableModel();
   for (const FiltersModel::Filter & filter : _filtersModel) {
+    if (filter.absolutePathNoTags().contains("About") && filter.absolutePathNoTags().contains("Support Us")) continue;
     if (filter.matchKeywords(keywords)) {
       _filtersView->addFilter(filter.name(), filter.hash(), filter.path(), filter.isWarning());
     }
diff --git a/src/HeadlessProcessor.cpp b/src/HeadlessProcessor.cpp
index de6dc90..53a141f 100644
--- a/src/HeadlessProcessor.cpp
+++ b/src/HeadlessProcessor.cpp
@@ -267,7 +267,7 @@ void HeadlessProcessor::endApplication(const QString & errorMessage)
   if (!errorMessage.isEmpty()) {
     Logger::error(errorMessage);
   }
-  QCoreApplication::exit(!errorMessage.isEmpty());
+//  QCoreApplication::exit(!errorMessage.isEmpty());
 }
 
 } // namespace GmicQt
