diff --git a/src/HtmlTranslator.cpp b/src/HtmlTranslator.cpp
index 1421fc9..f4f68ac 100644
--- a/src/HtmlTranslator.cpp
+++ b/src/HtmlTranslator.cpp
@@ -56,6 +56,7 @@ bool HtmlTranslator::hasHtmlEntities(const QString & str)
 
 QString HtmlTranslator::fromUtf8Escapes(const QString & str)
 {
+  if (str.isEmpty()) return str;
   QByteArray ba = str.toUtf8();
   gmic_library::cimg::strunescape(ba.data());
   return QString::fromUtf8(ba);
